File: JavaScript/src/Internals/Functions/MathTrigonometric/Function_COSH.js
import { Function_1 } from '../Function_1.js';
import { Operand } from '../../../Operand.js';

class Function_COSH extends Function_1 {
    get Name() {
        return "Cosh";
    }

    constructor(a) {
        super(a);
    }

    Evaluate(work, tempParameter) {
        let args1 = this.GetNumber_1(work, tempParameter);
        if (args1.IsError) { return args1; }
        return Operand.Create(Math.cosh(args1.DoubleValue));
    }
}

export { Function_COSH };

File: java/src/main/java/toolgood/algorithm/internals/functions/mathtrigonometric/Function_COSH.java
package toolgood.algorithm.internals.functions.mathtrigonometric;

import java.lang.StringBuilder;
import java.util.function.Function;
import toolgood.algorithm.internals.Operand;
import toolgood.algorithm.internals.AlgorithmEngine;
import toolgood.algorithm.internals.functions.Function_1;
import toolgood.algorithm.internals.functions.FunctionBase;

class Function_COSH extends Function_1 {
    public Function_COSH(FunctionBase func1) {
        super(func1);
    }

    @Override
    public Operand Evaluate(AlgorithmEngine work, Function<AlgorithmEngine, String, Operand> tempParameter) {
        Operand args1 = func1.Evaluate(work, tempParameter);
        if (!args1.isNumber()) {
            args1 = args1.toNumber("Function '{0}' parameter is error!", "Cosh");
            if (args1.isError()) {
                return args1;
            }
        }
        return Operand.Create(Math.cosh(args1.getDoubleValue()));
    }

    @Override
    public void toString(StringBuilder stringBuilder, boolean addBrackets) {
        AddFunction(stringBuilder, "Cosh");
    }
}
